spec/services/problem_check/starttls_disabled_spec.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+RSpec.describe ProblemCheck::StarttlsDisabled do
+  subject(:check) { described_class.new }
+
+  describe ".call" do
+    context "with STARTTLS enabled" do
+      before { GlobalSetting.stubs(:smtp_enable_start_tls).returns(true) }
+      it { expect(check).to be_chill_about_it }
+    end
+
+    context "with STARTTLS disabled" do
+      before { GlobalSetting.stubs(:smtp_enable_start_tls).returns(false) }
+      it do
+        expect(check).to have_a_problem.with_priority("high").with_message(
+          I18n.t("dashboard.problem.#{check.identifier}"),
+        )
+      end
+    end
```

Comment:
would a test for a nil value be good to have?

Reply:
I don't think so? I'm mimicking other checks here, based off the [example](https://github.com/discourse/discourse-translator/pull/155) from the [docs](https://dev.discourse.org/t/problem-checks-and-how-to-use-them/128444)